src/main/java/cubicchunks/lighting/FirstLightProcessor.java

```diff
@@ -74,6 +75,18 @@ public FirstLightProcessor(GeneratorStage lighting, String name, ICubeCache cach
 		ICubicWorld world = cube.getWorld();
 		ICubeCache cache = world.getCubeCache();
 
+		for (int x = -2; x <= 2; ++x) {
+			for (int y = -2; y <= 2; ++y) {
+				for (int z = -2; z <= 2; ++z) {
+					CubeCoords coords = new CubeCoords(cube.getX() + x, cube.getY() + y, cube.getZ() + z);
+					if (!cache.cubeExists(coords) || cache.getCube(coords).getCurrentStage().precedes(generatorStage)) {
+						CubicChunks.LOGGER.error("Missing cube ({})! {}/{}", coords, cache.getCube(coords).getCurrentStage().getName(), generatorStage.getName());
+						System.exit(1);
```

Comment:
Ok, just keep in mind that before merging it, `System.exit()` must be removed. Or possibly remove this check entirely, and then check that `world.checkLightFor()` returns `true`.